File: bridge/core/html/image.cc
/*
 * Copyright (C) 2022-present The WebF authors. All rights reserved.
 */

#include "image.h"

namespace webf {

Image* Image::Create(ExecutingContext* context, ExceptionState& exception_state) {
  return MakeGarbageCollected<Image>(context, exception_state);
}

Image::Image(ExecutingContext* context, ExceptionState& exception_state) : HTMLImageElement(*context->document()) {}

}  // namespace webf

File: webf/lib/src/rendering/image.dart
/*
 * Copyright (C) 2024-present The OpenWebF Company. All rights reserved.
 * Licensed under GNU GPL with Enterprise exception.
 */
/*
 * Copyright (C) 2019-2022 The Kraken authors. All rights reserved.
 * Copyright (C) 2022-2024 The WebF authors. All rights reserved.
 */
import 'package:flutter/rendering.dart';
import 'package:flutter/widgets.dart';

class WebFRenderImage extends RenderImage {
  WebFRenderImage({
    super.image,
    super.fit,
    super.alignment,
  });

}

class WebFRawImage extends RawImage {
  const WebFRawImage({
    super.key,
    super.image,
    super.debugImageLabel,
    super.width,
    super.height,
    super.scale = 1.0,
    super.color,
    super.opacity,
    super.colorBlendMode,
    super.fit,
    super.alignment
  });

  @override
  RenderImage createRenderObject(BuildContext context) {
    return WebFRenderImage(image: image?.clone(), fit: fit, alignment: alignment);
  }
}
